docs/dev.md

```diff
@@ -21,75 +21,40 @@ the loop of iterating and re-deploying small changes, you will definitely prefer
-uses the tag to record the version.
-`git tag "v1.2.3"`
-
-Create a new
-[PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token)
-in the github UI, ensuring that it has "package:write"
-permissions. After you click "package:write" the other necessary permissions
-will auto-select. You can set a 7-day expiration, but you should delete the
-token after the release for maximum security.
-
-`export CR_PAT=<your-token-here>`
-and then run a docker login:
-`echo $CR_PAT | docker login ghcr.io -u <your-username-here>  --password-stdin`
-
-Ensure that you've linked your command-line to your kubernetes environment:
-`eval $(minikube docker-env)` (or similar if you're using kind, etc.)
-
-Run the release script:  `hack/build-release.sh`
-
-Push the tag `git push --tags origin HEAD` and then in the github UI make a new
-release from that tag.
-In the release, add a triple-tick section at the bottom with the checksum that
-was output by the release script and the filename (release.yml, package.yml and metadata.yml,  omit the
-`./tmp` prefix). Ensure there's two spaces between the checksum and the filename
-to make it interoperable with checksum tooling verification.
-
-In your OS UI, open the secretgen-controller repo tmp folder and drag the
-release.yml, package.yml and metadata.yml assets into the github UI release as attached files.
-
-Test the release artifact to ensure it actually works
-`kapp deploy -a secretgen -f releases/1.2.3.yml`
-Ensure the deployment and build worked by running the e2e tests:
-you may need to `export SECRETGEN_E2E_NAMESPACE=secretgen-test`
-then `./hack/test-e2e.sh`.
-
-Test the package artifact to ensure it actually works
-`kapp delete -a secretgen`
-
-Create a PackageInstall file (pkgi.yml) which will be used to install of Secretgen-controller package
+Release versions are scraped from git tags in the same style as the goreleaser
+tool.
 
+Tag the release - it's necessary to do this first because the release process uses the latest tag to record the version.
 ```
----
-apiVersion: packaging.carvel.dev/v1alpha1
-kind: PackageInstall
-metadata:
-  name: secretgen-controller
-spec:
-  serviceAccountName: cluster-admin-sa
-  packageRef:
-    refName: secretgen-controller.carvel.dev
-    apiVersion: packaging.carvel.dev/v1alpha1
-    versionSelection:
-      constraints: ">0.8.0"
+git tag "v1.2.3"
 ```
 
-> Note: cluster-admin-sa comes from deploying https://github.com/vmware-tanzu/carvel-kapp-controller/tree/develop/examples/rbac
-
-`kapp deploy -a secretgen-package -f package.yml -f metadata.yml -f pkgi.yml`
-Ensure the deployment and build worked by running the e2e tests:
-you may need to `export SECRETGEN_E2E_NAMESPACE=secretgen-test`
-then `./hack/test-e2e.sh`.
-
-After verifying, commit: `git add releases; git commit -m "release 1.2.3"`
-and push: `git push origin HEAD --tags`
+Push the tag to GitHub.
+```
+git push --tags
```

Comment:
Should we suggest `git push origin v1.2.3` instead? (`git push --tags` would push all the tags present in local which might not be desired)

Reply:
sure thing, just took this verbatim from the kapp-controller doc. I like this suggestion though 👍